src/cart/utils/crt_utils.c

```diff
@@ -346,12 +349,141 @@ crtu_load_group_from_file(const char *grp_cfg_file, crt_context_t ctx,
 	return rc;
 }
 
+#define SYS_INFO_BUF_SIZE 16
+
+int
+crtu_dc_mgmt_net_cfg_rank_add(const char *name, crt_group_t *group,
+		    crt_context_t *context)
+{
+	int	i;
+	int rc = 0;
+	struct dc_mgmt_sys_info crt_net_cfg_info;
+	Mgmt__GetAttachInfoResp *crt_net_cfg_resp;
+
+	/* Query the agent for the CaRT network configuration parameters */
+	rc = get_attach_info(name, true /* all_ranks */, &crt_net_cfg_info, &crt_net_cfg_resp);
+	D_ASSERTF(rc == 0, "get_attach_info failed, rc=%d\n", rc);
```

Comment:
we have opts.assert_on_error (see line 47) that each test sets. we should only assert if its true

Reply:
Ah, I see.  So this option allows us to force tests to proceed despite an error?  We currently set `assert_on_error` to `true` for all tests in [tests/ftest/cart](https://github.com/daos-stack/daos/blob/master/src/tests/ftest/cart).  I assume we want the `opts.assert_on_error` condition for all `D_ASSERTF`'s, so I'll add it to other places in `crt_utils.c` where it's missing.